src/main/java/Duke/Duke.java

```diff
@@ -0,0 +1,174 @@
+            if (command == Commands.BYE) {
+                break;
+            } else {
+                executeCommand(command, input);
+            }
+        }
+        ui.showGoodbyeMessage();
+    }
+
+
+    /**
+     * Processes and runs user commands.
+     *
+     * @param command user command
+     * @param input   input string
+     */
+    private void executeCommand(Commands command, String input) {
+        if (command == Commands.LIST) {
+            System.out.println("    ____________________________________________________________\n");
+            for (int i = 1; i <= tasks.getSize(); i++) {
+                String index = String.format("%d.", i);
+                String entry = index + tasks.getItem(i).toString();
+                ui.showEntry(entry);
+
+            }
+            System.out.println("    ____________________________________________________________\n");
+
+        } else if (command == Commands.MARK) {
+            String number = input.split(" ", 2)[1];
+            try {
+                int num = Integer.parseInt(number);
+                tasks.markTasks("mark", num);
+                storage.rewriteFile(tasks.getTaskArray());
+            } catch (NumberFormatException e) {
+                ui.showInvalidTaskMessage();
+            } catch (IOException e) {
+                ui.showErrorWritingMessage();
+            }
+        } else if (command == Commands.UNMARK) {
+            try {
+                String number = input.split(" ", 2)[1];
+                int num = Integer.parseInt(number);
+                tasks.markTasks("unmark", num);
+                storage.rewriteFile(tasks.getTaskArray());
+            } catch (NumberFormatException e) {
+                ui.showInvalidTaskMessage();
+            } catch (IOException e) {
+                ui.showErrorWritingMessage();
+            }
+        } else if (command == Commands.DELETE) {
+            String number = input.split(" ", 2)[1];
+            try {
+                int num = Integer.parseInt(number);
+                Task toDelete = tasks.getTaskArray().get(num - 1);
+                tasks.delete(toDelete);
+                ui.showDeleteMessage(toDelete, tasks.getSize());
+                storage.writeToFile(toDelete);
+            } catch (NumberFormatException e) {
+                ui.showInvalidTaskMessage();
+            } catch (IndexOutOfBoundsException e) {
+                ui.showInvalidIndexMessage();
+            } catch (IOException e) {
+                ui.showErrorWritingMessage();
+            }
+        } else if (command == Commands.CREATETASK) {
+            try {
+                Task newTask = tasks.createTask(input, false);
+                if (newTask != null) {
+                    storage.writeToFile(newTask);
+                }
+            } catch (DukeException.EmptyTaskException | DukeException.UnkownCommandException | DukeException.InvalidParameterException error) {
```

Comment:
Line might be too long, could try to wrap the last exception into the next line!